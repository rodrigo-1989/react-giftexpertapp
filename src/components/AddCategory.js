import React,{ useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategorie  } ) => {
    const [inputValue,setInputValue ]=useState(''); 
    
    const handleInputChange = (e)=>{
           setInputValue( e.target.value );      
    }
    const handleSubmit =(e) =>{
        e.preventDefault();
        if (inputValue.trim().length > 2){
            setCategorie(cats => [inputValue,...cats]);
            setInputValue( '' ); }
    }
    return (
        <form onSubmit= { handleSubmit }>
            <input type="text"
                value= { inputValue }
                onChange= { handleInputChange }/>
            
        </form>
    )
}
AddCategory.propTypes = {
    setCategorie: PropTypes.func.isRequired
}
