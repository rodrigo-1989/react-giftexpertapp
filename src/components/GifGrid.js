import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
export const GifGrid = ({ category }) => { 

       const { data:images } =  useFetchGifs( category );
    return (
        <>
        <h3>{ category }</h3>
       
            <div className="card-grid"> 
                    {
                    images.map( img =>(
                        <GifGridItem key = { img.id }  { ...img }/>
                    ))
                    }
            </div>
        </>
    )
}
 
GifGrid.propTypes={
    category: PropTypes.string.isRequired
}