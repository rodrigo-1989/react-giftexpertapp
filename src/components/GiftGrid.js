import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'

 import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({ category }) => { 

       const { data:images } =  useFetchGifs( category );
    return (
        <>
        <h3>{ category }</h3>
       
            <div className="card-grid"> 
                    {
                    images.map( img =>(
                        <GiftGridItem key = { img.id }  { ...img }/>
                    ))
                    }
            </div>
        </>
    )
}
 