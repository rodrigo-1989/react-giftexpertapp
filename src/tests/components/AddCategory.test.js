import   { shallow  }  from  'enzyme';
import { AddCategory } from '../../components/AddCategory';
import React from 'react';

describe('Pruevas en el componente AddCategory', () => {
    const setCategorie = jest.fn();
    let wraper = shallow(<AddCategory setCategorie = { setCategorie }/>);

    //ciclo de vida para cada test 
    beforeEach(()=>{
        jest.clearAllMocks();
         wraper = shallow(<AddCategory setCategorie = { setCategorie }/>);
    });

    test('Debe de mostrarse correctamente', () => {
        expect( wraper ).toMatchSnapshot();
        
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wraper.find('input');

        const value = 'Hola mundo';
        input.simulate( 'change' , { target:{ value } });

        expect( wraper.find('p').text().trim() ).toBe( value );
    });

    test('No debe de postaer la informacion con el submit ', () => {
        wraper.find('form').simulate('submit',{ preventDefault(){}});
        expect( setCategorie ).not.toHaveBeenCalled(); 
        
    });
    
    test('Debe de llamar setCategorie y limpiar la caja de texto', () => {
        const value = 'Hola Mundo';
        //Simular el input change
        wraper.find('input').simulate( 'change' , { target:{ value } });
        //simular submit
        wraper.find('form').simulate('submit',{ preventDefault(){}});
        //Setcategorie debe ser llamado
        expect( setCategorie ).toHaveBeenCalled(); 
        //ver que la caja de texto quede limpia ''
        expect( wraper.find('input' ).prop( 'value' ) ).toBe( '' );
    })
    
    
    
})
