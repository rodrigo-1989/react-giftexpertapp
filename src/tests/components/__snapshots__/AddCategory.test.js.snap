// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en el componente AddCategory Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
